src/gmpv_control_box.c

```diff
@@ -511,6 +518,10 @@ static void gmpv_control_box_init(GmpvControlBox *box)
 				"clicked",
 				G_CALLBACK(simple_signal_handler),
 				box );
+g_signal_connect(   box->repeat_button,
```

Comment:
Instead of using click signal here, it would be better to create a new property and bind it with the  `loop-playlist` property in GmpvModel. We can then bind the repeat button's `active` property to this property.